### PR TITLE
fix(i18n): pass search params to fallback

### PR DESCRIPTION
**Pull request by mtwilliams-code**
## Changes

- makes the i18n middleware when used for fallbacks pass on the search params to the subsequent render.
- closes issue #12546 

## Testing

Tested locally with the same code in the stackblitz example in issue. Ran all i18n test suites. Did not add a specific test for this - if maintainers want I will, but wanted feedback on willingness to include this before I dove into the fixtures.

## Docs

/cc @withastro/maintainers-docs for feedback!
May need docs. The current behavior is also undocumented and imo less intuitive.

### CODE REVIEW

**Comment by ascorbic** [comment]
This looks good. Could you add a test for this? Probably in `packages/astro/test/i18n-routing.test.js`

**Comment by mtwilliams-code** [comment]
@ematipico this seems to be your baby - any concerns?

File: .changeset/kind-mayflies-argue.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
+---
+
+pass search params to i18n fallbacks
```
Comment (ematipico) [inline .changeset/kind-mayflies-argue.md:5]:
```suggestion
Fixes a bug where URL search parameters weren't passed when using the i18n `fallback` feature.
```

File: packages/astro/test/i18n-routing.test.js [hunk 2]
```diff
@@ -1582,6 +1582,22 @@ describe('[SSR] i18n routing', () => {
 			assert.equal(response.status, 404);
 		});
 
+		it('should pass search to render when using requested locale', async () => {
+			let request = new Request('http://example.com/new-site/pt/start?search=1');
+			let response = await app.render(request);
+			assert.equal(response.status, 200);
+			const text = await response.text();
+			assert.equal(text.includes('Oi essa e start'), true);
+			assert.equal(text.includes('search=1'), true);
```
Comment (ematipico) [inline packages/astro/test/i18n-routing.test.js:1591]:
```suggestion
			assert.match(text, /Oi essa e start/);
			assert.match(text, /search=1/);
```

File: .changeset/kind-mayflies-argue.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
```
Comment (ematipico) [inline .changeset/kind-mayflies-argue.md:2]:
```suggestion
'astro': patch
```

**Review by ematipico — CHANGES_REQUESTED** [review]
The fixes look good, we need to address the changeset, it's not a minor (this is a blocker)

I also, I suggested few changes in the tests.

**Review by ematipico — APPROVED** [review]

**Pull request merged**